packages/gatsby-theme-mdx/package.json

```diff
@@ -17,7 +17,8 @@
   },
   "peerDependencies": {
     "react": ">=16",
-    "react-dom": ">=16"
+    "react-dom": ">=16",
+    "gatsby": "^2.0.0"
```

Comment:
This doesn't seem to be front matter related?
And also the existing/previous behavior is how Gatsby documents how themes should be made https://www.gatsbyjs.com/tutorial/building-a-theme/

Reply:
https://www.gatsbyjs.com/tutorial/building-a-theme/#add-development-dependencies-to-gatsby-theme-events

Reply:
It results conflict gastby versions when I tried to upgrade gastby to v3 to test if it’s compatible with remark-mdx@2 now.

Reply:
Gotcha, thanks for the clarification

It feels like this PR now captures three distinct change sets, fixing the frontmatter docs, adjusting the eslint settings, and updates to the documentation site.
All are good and welcome.

Reviewing them all together, especially when the title and description has no mention of the latter to is a bit confusing